dpc-api/src/test/java/gov/cms/dpc/api/resources/v1/PatientResourceUnitTest.java

```diff
@@ -2,38 +2,119 @@
+import org.hl7.fhir.instance.model.api.IBaseOperationOutcome;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.Answers;
 import org.mockito.Mock;
-import org.mockito.Mockito;
-import org.mockito.MockitoAnnotations;
 
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.WebApplicationException;
 import javax.ws.rs.core.Response;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
 import java.util.UUID;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
+import static gov.cms.dpc.api.APITestHelpers.createProvenance;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.openMocks;
 
 public class PatientResourceUnitTest {
 
     @Mock(answer = Answers.RETURNS_DEEP_STUBS)
     IGenericClient attributionClient;
 
-    @Mock
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
     FhirValidator fhirValidator;
 
-    PatientResource resource;
+    @Mock
+    DataService dataService;
+
+    @Mock
+    BlueButtonClient bfdClient;
+
+    PatientResource patientResource;
 
     @BeforeEach
     public void setUp() {
-        MockitoAnnotations.openMocks(this);
-        resource = new PatientResource(attributionClient, fhirValidator, null, null);
+        openMocks(this);
+        patientResource = new PatientResource(attributionClient, fhirValidator, dataService, bfdClient);
+    }
+
+    @Test
+    public void testPatientSearch() {
+        UUID orgId = UUID.randomUUID();
+        Organization organization = new Organization();
+        organization.setId(orgId.toString());
+        OrganizationPrincipal organizationPrincipal = new OrganizationPrincipal(organization);
+        String patientMbi = "3aa0C00aA00";
+        String expandedMbi = String.format("%s|%s", DPCIdentifierSystem.MBI.getSystem(), patientMbi);
+        Patient patient = new Patient();
+        patient.addIdentifier().setSystem(DPCIdentifierSystem.MBI.getSystem()).setValue(patientMbi);
+        patient.setManagingOrganizationTarget(organization);
+        Bundle bundle = new Bundle();
+        bundle.addEntry().setResource(patient);
+
+        @SuppressWarnings("unchecked")
+        IQuery<IBaseBundle> queryExec = mock(IQuery.class, Answers.RETURNS_DEEP_STUBS);
+        @SuppressWarnings("unchecked")
+        IQuery<Bundle> mockQuery = mock(IQuery.class);
+        when(attributionClient
+                .search()
+                .forResource(Patient.class)
+                .encodedJson()
+        ).thenReturn(queryExec);
+        when(queryExec.where(Patient.ORGANIZATION.hasId(orgId.toString())).returnBundle(Bundle.class)).thenReturn(mockQuery);
+        when(mockQuery.where(Patient.IDENTIFIER.exactly().identifier(expandedMbi))).thenReturn(mockQuery);
```

Comment:
I think the test failures are due to the Object equality matching failing here. When I replace with generic matching the test passes:

```suggestion
        when(queryExec.where(any(ICriterion.class)).returnBundle(Bundle.class)).thenReturn(mockQuery);
        when(mockQuery.where(any(ICriterion.class)).thenReturn(mockQuery);
```

It reduces the test effectiveness a bunch but I notice the ConsentServiceImplUnitTest does the same type of generic matching.

I thought maybe you could use argument capturing to validate that the provided orgId/expandedMbi is correct, but ICriterion doesn't have any public interface methods for getting information out of it ☹️ 

```
ArgumentCaptor<ICriterion> argument = ArgumentCaptor.forClass(ICriterion.class);
when(queryExec.where(argument.capture())).returnBundle(...)

ICriterion criteria = argument.getValue();
// can't retrieve any attributes from criteria...
```